### ISSUE TITLE
Cannot serialize enums

### ISSUE THREAD

**Issue by lhoste-bell** [nokia/ts-serialize-closures#4]
Enums are a JS/TS construct which we don't properly serialize yet. Consider the following unit test:

```
function roundtrip<T>(value: T): T {
  return deserialize(serialize(value));
}

export enum Match {
  CONTAINS = "CONTAINS",
  CROSSES = "CROSSES",
  DISJOINT = "DISJOINT"
}

export function enumString() {
  expect(roundtrip(Match.CROSSES)).to.equal("CROSSES");
}

```

**Comment by jonathanvdc** [comment]
Interesting. The reason that unit test doesn't work appears to be the fact that `Match` is exported, not so much that fact that `Match` is an enum.

Indeed, if we remove the `export` keyword from `Match`'s definition, then TypeScript compiles `Match` down to the following code:

```ts
var Match;
(function (Match) {
    Match["CONTAINS"] = "CONTAINS";
    Match["CROSSES"] = "CROSSES";
    Match["DISJOINT"] = "DISJOINT";
})(Match || (Match = {}));
```

Since the snippet above does not capture any values and does not contain any variables that are defined more than once (`Match`'s unique definition is `Match = {}`), the serializer does not in any way transform `tsc`'s output for `Match`.

When we restore the `export` keyword, `tsc`'s codegen doesn't change much.

```ts
var Match;
(function (Match) {
    Match["CONTAINS"] = "CONTAINS";
    Match["CROSSES"] = "CROSSES";
    Match["DISJOINT"] = "DISJOINT";
})(Match = exports.Match || (exports.Match = {}));
```

But for some reason the serializer goes haywire here and transforms the above into the following:

```ts
var Match = { value: undefined };
(function (Match) {
    Match["CONTAINS"] = "CONTAINS";
    Match["CROSSES"] = "CROSSES";
    Match["DISJOINT"] = "DISJOINT";
})(Match.value = exports.Match.value || (exports.Match.value = {}));
```

This codegen is erroneous; the serializer appears to conflate `Match` and `exports.Match`, triggering inappropriate assignment conversion. I strongly suspect that this is due to the way the TypeScript compiler represents compiler-generated `exports.Match` expressions. Specifically, I think `tsc` may be representing `exports.Match` as a single identifier expression rather than a property access expression.

I'll see what I can do to make the serializer distinguish `tsc`'s representation of `Match` expressions from compiler-generated `exports.Match` expressions.

In the meantime, you can use the following idiom, which does not appear to confuse the serializer:

```ts
enum Match {
    CONTAINS = "CONTAINS",
    CROSSES = "CROSSES",
    DISJOINT = "DISJOINT"
}
exports.Match = Match;
```

Indeed, the serializer produces the following code for the above:

```ts
var Match;
(function (Match) {
    Match["CONTAINS"] = "CONTAINS";
    Match["CROSSES"] = "CROSSES";
    Match["DISJOINT"] = "DISJOINT";
})(Match || (Match = {}));
exports.Match = Match;
```

**Comment by jonathanvdc** [comment]
Alright, I submitted a fix for this issue: #5.
